### PR TITLE
Refactor permissions by using new enum OrganizationPermission

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-db-dao/src/main/java/org/sonar/core/permission/GlobalPermissions.java [hunk 1]
```diff
@@ -26,7 +26,9 @@
 /**
  * Holds the constants representing the various global permissions that can be assigned to users & groups
  *
+ * @deprecated replaced by enum {@link org.sonar.db.permission.OrganizationPermission}
  */
+@Deprecated
 public final class GlobalPermissions {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/core/permission/GlobalPermissions.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Comment by simonbrandhof** [comment]
Many DB tests still rely on {{GlobalPermissions}}. They should be refactored but I can't do that for the time being. Goal will be to drop {{GlobalPermissions}} and then to introduce the equivalent enum {{ProjectPermission}}.

File: server/sonar-server/src/main/java/org/sonar/server/permission/OrganizationPermission.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    return key;
+  }
+
+  public static OrganizationPermission fromKey(String key) {
+    for (OrganizationPermission p : values()) {
+      if (p.getKey().equals(key)) {
+        return p;
+      }
+    }
+    throw new IllegalArgumentException("Unsupported permission: " + key);
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/permission/OrganizationPermission.java]: s/permission/permission key ?

File: server/sonar-server/src/test/java/org/sonar/server/organization/ws/SearchMyOrganizationsActionTest.java [hunk 3]
```diff
@@ -159,8 +161,8 @@ public void returns_organization_of_authenticated_user_only_for_ADMIN_permission
     OrganizationDto organization2 = dbTester.organizations().insert();
     GroupDto group = dbTester.users().insertGroup(organization2);
     dbTester.users().insertMember(group, user);
-    GlobalPermissions.ALL.stream()
-      .filter(s -> !s.equals(SYSTEM_ADMIN))
+    Arrays.stream(OrganizationPermission.values())
```
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/organization/ws/SearchMyOrganizationsActionTest.java]: could be worth having a method `OrganizationPermission.all()` which returns a `Stream<OrganizationPermission>`, WDYT?

`GlobalPermissions.ALL` was used in several places

File: server/sonar-db-dao/src/main/java/org/sonar/core/permission/GlobalPermissions.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-db-dao/src/main/java/org/sonar/core/permission/GlobalPermissions.java]: commit title: `Deprecated GlobalPermissions` => `Deprecate GlobalPermissions`

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java [hunk 4]
```diff
@@ -246,6 +246,6 @@ public RulesProfile importProfile(Reader reader, ValidationMessages messages) {
   private void logInAsQProfileAdministrator() {
     userSession
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: No, `defaultOrganizationProvider.get()` does not return a `OrganizationDto`.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CopyActionTest.java [hunk 5]
```diff
@@ -153,6 +153,6 @@ public void throw_UnauthorizedException_if_not_logged_in() throws Exception {
   private void logInAsQProfileAdministrator() {
     userSessionRule
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CopyActionTest.java]: No, defaultOrganizationProvider.get() does not return a OrganizationDto.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java [hunk 6]
```diff
@@ -190,6 +190,6 @@ public void fail_if_profile_does_not_exist() throws Exception {
   private void logInAsQProfileAdministrator() {
     userSessionRule
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: No, defaultOrganizationProvider.get() does not return a OrganizationDto.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RenameActionTest.java [hunk 7]
```diff
@@ -164,6 +164,6 @@ private void createProfiles() {
   private void logInAsQProfileAdministrator() {
     userSessionRule
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RenameActionTest.java]: same

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreActionTest.java [hunk 8]
```diff
@@ -115,6 +115,6 @@ public void throw_UnauthorizedException_if_not_logged_in() throws Exception {
   private void logInAsQProfileAdministrator() {
     userSessionRule
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreActionTest.java]: same :-)

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java [hunk 9]
```diff
@@ -90,6 +90,6 @@ public void throw_UnauthorizedException_if_not_logged_in() throws Exception {
   private void logInAsQProfileAdministrator() {
     userSession
       .logIn()
-      .addOrganizationPermission(defaultOrganizationProvider.get().getUuid(), GlobalPermissions.QUALITY_PROFILE_ADMIN);
+      .addPermission(ADMINISTER_QUALITY_PROFILES, defaultOrganizationProvider.get().getUuid());
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInActionTest.java]: same :-)

File: server/sonar-server/src/test/java/org/sonar/server/organization/ws/SearchMyOrganizationsActionTest.java (hunk 3 above)
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/organization/ws/SearchMyOrganizationsActionTest.java]: not so many times in fact. But why not.

**Pull request closed without merging**